### ISSUE TITLE
3.2 Cross Validation

### ISSUE THREAD

**Issue by MR-NBD** [MR-NBD/ML4N--Adversarial-Attacks-Polito-2023-24#7]
# Cross Validation
Implement a rigorous cross-validation process involving training, evaluating and tuning of the hyperparameters to get the best validation accuracy. Each classifier’s performance should be summarized with a brief description of the optimized hyperparameters and a rationale for observed outcomes.
